engine.py

```diff
@@ -108,6 +116,57 @@ def post_app_init(self):
             except:
                 traceback.print_exc()
 
+    def validate_context(self, tank, context):
+        '''
+        Given the context provided at initialisation, if we're in a project context only, forces the user to select
+        a proper task context using rdokatana.taskChooser.taskChooser.TaskChooser.
+        :param tank: a Tank object instance.
+        :type tank: :class:`sgtk.tank.Tank`
+        :param context: the current context.
+        :type context: :class:`tank.context.Context`
+        :return: a proper working context.
+        :rtype: :class:`tank.context.Context`
+        '''
+        if context.project:
+            if not context.entity:
+                task = self.select_task_ui(context)
+                newContext = tank.context_from_entity('Task', task['id'])
+                return newContext
+            elif context.entity and not context.task:
+                filters = [
+                    ['project', 'is', context.project],
+                    ['entity', 'is', context.entity],
+                ]
+                task = None
+                if context.entity['type'] == 'Shot':
+                    filters.append(['content', 'is', 'Lighting'])
+                    task = tank.shotgun.find_one("Task", filters)
+                elif context.entity['type'] == 'Asset':
+                    filters.append(['content', 'is', 'Shading'])
+                    task = tank.shotgun.find_one("Task", filters)
+                if task:
+                    newContext = tank.context_from_entity('Task', task['id'])
+                    return newContext
+                else:
+                    task = self.select_task_ui(context)
+                    newContext = tank.context_from_entity('Task', task['id'])
+                    return newContext
```

Comment:
Users can't launch Katana with a project context? If so, I find it a bit annoying to not be able to use a project context. I find this really restrictive. But I guess it was a request from someone.

For point 2, this is something we wanted to do for a long time. Having a default selected task when launching a DCC with an entity is a great feature. So as a note for the future, we should start to think a about a global solution that could be used by every DCC at some point.